### DISCUSSION TITLE
Only the first item in the RSS is considered?

### DISCUSSION THREAD

**Discussion by cregecambrin** [vslavik/winsparkle#270] [Q&A] [answered]
Hi, I tried setting up an RSS feed with multiple items, one for each available version. It looks like Winsparkle only considers the first item.
Some examples:

If my app is on version 1.2, and the feed has three entries, 1.1, 1.2 and 1.3, checking for updates returns "You're up to date". I would expect it to offer me to upgrade to 1.3 instead

If my app is on version 1.0, and the feed has three entries, 1.1, 1.2 and 1.3 (in that order), checking for updates returns "Version 1.1 is available" and the release notes for 1.1 are shown. I would expect it to offer me to upgrade to 1.3 instead, and the Release notes section to show a collection of the release notes for 1.1, 1.2 and 1.3

Am I missing something, or this the expected behavior?

**Comment by vslavik** [comment, marked as answer]
Sane RSS feeds are sorted in reverse chronological order. In principle, feed parsing should consider pubDate if present, and contributions to that effect are welcome, but in practice, it is a non-issue (well, obviously with the exception of your feed).

Edit: Sparkle [picks the newest item from the feed too](https://github.com/sparkle-project/Sparkle/blob/1b6565ded582f2c7c3ec10a7d917cb2ed4911c3c/Sparkle/SUAppcastDriver.m#L402), so while sorting by date and/or version is doable, it's probably not a good idea to do.

Edit 2: See #273.

**Reply by cregecambrin** [reply]
Thank you for the quick answer.

I have no problem ordering the feed in reverse order. However, to put in another use case:
if my app is on version 1.0, and the feed has three entries, 1.3, 1.2 and 1.1 (in that order), checking for updates returns "Version 1.3 is available" (as expected) and the release notes for 1.3 are shown, but not the ones for 1.2 and 1.1. Can you confirm this is expected behavior?

**Reply by vslavik** [reply]
Appropriate release notes page is your responsibility.
